magefiles/docs.go

```diff
@@ -26,4 +39,131 @@ func main() {
 	if err := doc.GenMarkdownTree(cmd, "./docs/docs/references/configuration/cli"); err != nil {
 		log.Fatal("Fatal error", log.Err(err))
 	}
+	if err := generateConfigDocs("./docs/docs/references/configuration/config-file.md"); err != nil {
+		log.Fatal("Fatal error in config file generation", log.Err(err))
+	}
+}
+
+// generateConfigDocs creates markdown file for Trivy config.
+func generateConfigDocs(filename string) error {
+	// remoteFlags should contain Client and Server flags.
+	// NewClientFlags doesn't initialize `Listen` field
+	remoteFlags := flag.NewClientFlags()
+	remoteFlags.Listen = flag.ServerListenFlag.Clone()
+
+	// These flags don't work from config file.
+	// Clear configName to skip them later.
+	globalFlags := flag.NewGlobalFlagGroup()
+	globalFlags.ConfigFile.ConfigName = ""
+	globalFlags.ShowVersion.ConfigName = ""
+	globalFlags.GenerateDefaultConfig.ConfigName = ""
+
+	var allFlagGroups = []flag.FlagGroup{
+		globalFlags,
+		remoteFlags,
```

Comment:
I think `Client/Server` flags are not most popular.
Perhaps we can move them to the end of the array or just lower in the list.

Reply:
I'm not sure about it. 
usually the content is searched in alphabetical order

Reply:
There are AWS options is in the end. 
I'm not sure, that we should add these flags now.

Reply:
> I think `Client/Server` flags are not most popular. Perhaps we can move them to the end of the array or just lower in the list.

Client/Server flags are reordered now

Reply:
> usually the content is searched in alphabetical order

Yes. You are right.
Overall, the alphabetical order suits me.

> I'm not sure, that we should add these flags now.

Why?

Reply:
> Why?

```sh
$ trivy aws
Trivy AWS is now available as an optional plugin. See github.com/aquasecurity/trivy-aws for details.
```

Reply:
oh... right! I forgot about that.
So we don't need to include `aws` flags.

Reply:
AWS flags removed